File: ClientApp/src/app/common/ChatMessage.ts

export class ChatMessage {
  fromUser?: string;
  toUser?: string;
  message?: string;
  timestamp?: string;
  index?: number;
  seen: boolean;
}

export class AudioMessage {
  fromUser?: string;
  toUser?: string;
  pcmStream: number[];
}

export class UserChat {
  userName: string
  chatMessages: ChatMessage[]
}

File: ChatAppPoc/Data/ChatMessage.cs
using ChatAppPoc.Models;
using System;
using System.ComponentModel.DataAnnotations;

namespace ChatAppPoc.Data
{
    public class ChatMessage
    {
        [Key]
        public virtual int Id { get; set; }

        public virtual string FromUser { get; set; }

        public virtual string ToUser { get; set; }

        public virtual string Message { get; set; }

        public virtual DateTime CreatedAt { get; set; }

        public bool Seen { get; set; }

        internal ChatMessageVm ToChatMessageVm()
        {
            return new ChatMessageVm
            {
                CreatedAt = CreatedAt,
                FromUser = FromUser,
                ToUser = ToUser,
                Message = Message,
                Index = Id,
                Timestamp = CreatedAt.ToString("t"),
                Seen = Seen
            };
        }
    }
}
